test/tests/functional/pbs_hook_execjob_end.py

```diff
@@ -457,3 +457,21 @@ def test_qrun_arrayjob_on_mom_breakdown(self):
         self.server.expect(JOB, {'job_state': 'B'}, jid)
         subjid_1 = j.create_subjob_id(jid, 1)
         self.common_steps(subjid_1, self.mom)
+
+    def test_mom_restart(self):
+        """
+        Test to restart mom while execjob_end hook is running
+        """
+        hook_name = "end_hook"
+        attr = {'event': 'execjob_end', 'enabled': 'True', 'alarm': '100'}
+        self.server.create_import_hook(hook_name, attr, self.hook_body)
+        j = Job(TEST_USER)
+        j.set_sleep_time(5)
+        jid = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+        self.mom.log_match("Job;%s;executed execjob_end hook" %
+                           jid, n=100, max_attempts=10,
+                           interval=2)
+        self.mom.restart()
```

Comment:
How do you make sure the mom is restarted successfully?
I think you can check for the log message that indicates mom is restarted.

Reply:
If mom restart fails.. the PTL framework will catch it and report error. Please have a look at the test Logs-before fix, attached in the description .

Reply:
Ok, I see PTL framework is taking care of it.